sw/device/lib/dif/dif_otbn.h

```diff
@@ -73,29 +73,30 @@ typedef enum dif_otbn_result {
 } dif_otbn_result_t;
 
 /**
- * Error codes received from the hardware.
+ * OTBN Errors
+ *
+ * OTBN uses a bitfield to indicate which errors have been seen. Multiple errors
+ * can be seen at the same time. This enum gives the indices of the different
+ * bits that may be set in the error bitfield.
  */
-typedef enum dif_otbn_err_code {
-  // Keep error codes in sync with the hardware!
-  /** No error occurred. */
-  kDifOtbnErrCodeNoError = 0x0,
+typedef enum dif_otbn_err_bit {
   /** Load or store to invalid address. */
-  kDifOtbnErrCodeBadDataAddr = 0x1,
+  kDifOtbnErrBitBadDataAddr = 0,
   /** Instruction fetch from invalid address. */
-  kDifOtbnErrCodeBadInsnAddr = 0x2,
+  kDifOtbnErrBitBadInsnAddr = 1,
   /** Call stack underflow or overflow. */
-  kDifOtbnErrCodeCallStack = 0x3,
+  kDifOtbnErrBitCallStack = 2,
   /** Illegal instruction execution attempted */
-  kDifOtbnErrCodeIllegalInsn = 0x4,
+  kDifOtbnErrBitIllegalInsn = 3,
   /** LOOP[I] related error */
-  kDifOtbnErrCodeLoop = 0x5,
+  kDifOtbnErrBitLoop = 4,
   /** Error seen in Imem read */
-  kDifOtbnErrCodeFatalImem = 0x80,
+  kDifOtbnErrBitFatalImem = 5,
   /** Error seen in Dmem read */
-  kDifOtbnErrCodeFatalDmem = 0x81,
-  /** Error seen in RF */
-  kDifOtbnErrCodeFatalReg = 0x82
-} dif_otbn_err_code_t;
+  kDifOtbnErrBitFatalDmem = 6,
+  /** Error seen in RF read */
+  kDifOtbnErrBitFatalReg = 7
```

Comment:
If these are going to be arbitrary, consecutive indices, consider eliding the explicit index numbers from the enum.

If they need to match up with hardware, you should include static asserts in the .c file.

Reply:
I think Greg's added the static asserts. These indices do have to match the hardware: maybe there's no harm making it explicit? (I don't have strong feelings)

Reply:
They do need to match up with the hardware, which is checked with static asserts.

As they need to match up with hardware I prefer the explicit indices (e.g. someone will open this file with the thought 'I need to add kDifOtbnErrBitFoo for bit number 8', the explicit numbering makes this a little easier).

Also whilst the bits are currently consecutive perhaps we'll want to make them non-consecutive for some reason.